docs/documentation/accessibility.md

```diff
@@ -0,0 +1,142 @@
+# Accessibility
+
+This page goes over information relevant to both developers and users relating to accessibility considerations.
+
+Accessibility is very important to consider when developing applications.
+Colorblindness impacts roughly 4% of the general population, meaning that if your application gets even a small user base, there is a high probability that it is being used by someone with colorblindness.
+
+Fortunately, there are ways as both developers and users to address these accessibility concerns, as detailed on this page.
+
+## Colorblindness accessibility considerations for developers
+
+If you would like to make your CLI more accessible for others, there are a few rules of thumb you can follow:
+
+### 1. Use Rich features over Click features
+
+If you are using colors inside your print statements and interactive elements, you can make your CLI more accessible with the following:
+
+- Replace `click.echo()` with `rich.print()`
+- Replace `click.confirm()` with `rich.prompt.Confirm.ask()`
+- Replace `click.prompt()` with `rich.prompt.Prompt.ask()`
+
+The reason why the Rich features are more accessible than the corresponding Click features is because of the `NO_COLOR` environment variable, which the Rich library uses to disables all color. This environment variable does not work with `click.style()`, however.
+
+!!! info
+    `NO_COLOR` is a _de facto_ standard for disabling color across a wide variety of terminal programs and frameworks.
+    You can read more [here](https://no-color.org/) about the standard.
+
+So, for example:
+
+- `Confirm.ask("[red]Are you sure?[/]")` is more accessible because it works with `NO_COLOR`.
+- `click.confirm(click.echo("Are you sure?", fg="red"))` is less accessible because it cannot be overridden by `NO_COLOR`.
+
+### 2. Use 4-bit ANSI colors
+
+The 4-bit ANSI color system is a set of 16 different colors implemented in effectively every terminal, and they are the most common way to set colors.
+These colors are **not** deterministic; different terminals use slightly different hex values for the ANSI colors. [Wikipedia has a full breakdown of all the variations in these colors](https://en.wikipedia.org/wiki/ANSI_escape_code#3-bit_and_4-bit)
+
+!!! note
+    **rich-click**'s logo references the ANSI colors! 😁
+
+There are 16 total ANSI colors: 8 base ANSI colors, with each one having a "bright" variant:
+
+- `black`
+- `red`
+- `green`
+- `yellow`
+- `blue`
+- `magenta`
+- `cyan`
+- `white`
+- `bright_black`
+- `bright_red`
+- `bright_green`
+- `bright_yellow`
+- `bright_blue`
+- `bright_magenta`
+- `bright_cyan`
+- `bright_white`
+
+Additionally, each one of these can be modified with `dim`, which in modern terminals just applies a change to the opacity of the color.
+In total, when counting `dim`, this gives developers 32 different colors that can be shown.
+
+Below is a quick script that renders all of these colors:
+
+```python
```

Comment:
It'd be nice to use a rich terminal screenshot here to embed the output in the docs, without needing to run the script :)